x/zkauth/ante/zkauth.go

```diff
@@ -152,7 +154,63 @@ func (zdf ZKAuthDeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 		return zdf.dfd.AnteHandle(ctx, tx, simulate, next)
 	}
 
-	// Case of zkauth msg, does nothing in this case
+	// Almost no different from the implementation of zdf.dfd.AnteHandle, uses ZKauthAddress for feePayer.
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	if addr := zdf.ak.GetModuleAddress(authtypes.FeeCollectorName); addr.Empty() {
+		return ctx, fmt.Errorf("fee collector module account (%s) has not been set", authtypes.FeeCollectorName)
```

Comment:
How about using sdkerrors.Wrap like elsewhere?

Reply:
This is not an error case for user input.
I can't find a suitable sdkerror, so how about leave it to a higher-level function?

Reply:
Surely, or If the result of this function is Empty(), other modules seem to panic.